pkg/apis/servicecatalog/types.go

```diff
@@ -127,10 +127,9 @@ type ServiceClass struct {
 	Plans         []ServicePlan
 	PlanUpdatable bool // Do we support this?
 
-	// OSB-specific
-	// OSBGUID is the identity of this object for use with the OSB API.
+	// ExternalID is the identity of this object for use with the OSB API.
 	// Immutable.
-	OSBGUID string
+	ExternalID string
```

Comment:
> On today's call we agreed to set metadata.Name to the GUID, and ExternalID to the GUID as well. This would allow us to twiddle the metadata.Name field to something human readable later if we could ever figure out how w/o losing the ExternalID/GUID value. We should add a DisplayName to hold the human readable name though.

I did not agree to that :)  But I have another idea...

We do not have precedent for anything that involves `Name` field munging in k8s, and I don't think we should try to design for it.  Here's another approach we could consider, which has [precedent](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L4106-L4112): add two write-only fields to the instance spec like so:

```go
type InstanceSpec struct {
  // other fields omitted

  ServiceClassDisplayName string
  ServicePlanDisplayName string
}
```

...these fields would only be present in the versioned API.  An admission controller could rewrite them during create to write the actual IDs of into the `ServiceClassName` and `PlanName` fields.  How does that sound?  I think this might still mess with patch if the name of the service has changed.

In any event, I don't think there's a reason to maintain this field once the semantics are changed for `Name` on this type if we do not have a specific need for it.